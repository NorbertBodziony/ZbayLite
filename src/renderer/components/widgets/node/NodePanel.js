import React from 'react'
import PropTypes from 'prop-types'
import * as R from 'ramda'

import Grid from '@material-ui/core/Grid'
import ExpansionPanel from '@material-ui/core/ExpansionPanel'
import ExpansionPanelSummary from '@material-ui/core/ExpansionPanelSummary'
import ExpansionPanelDetails from '@material-ui/core/ExpansionPanelDetails'
import { withStyles } from '@material-ui/core/styles'

import ExpandLessIcon from '@material-ui/icons/ExpandLess'

import NodeStatus from '../../../containers/widgets/node/NodeStatus'
import NodePanelDetails from './NodePanelDetails'

const styles = theme => ({
  expansionDetails: {
    paddingLeft: 3 * theme.spacing.unit,
    paddingRight: 2 * theme.spacing.unit,
    paddingTop: 0,
    paddingBottom: 1.5 * theme.spacing.unit
  },
  expander: {
    backgroundColor: theme.palette.primary.light,
    boxShadow: 'none'
  }
})

export const NodePanel = ({ classes }) => (
  <Grid container justify='center'>
    <Grid item xs>
      <ExpansionPanel square className={classes.expander}>
        <ExpansionPanelSummary expandIcon={<ExpandLessIcon />}>
          <NodeStatus />
        </ExpansionPanelSummary>
        <ExpansionPanelDetails className={classes.expansionDetails}>
          <NodePanelDetails />
        </ExpansionPanelDetails>
      </ExpansionPanel>
    </Grid>
  </Grid>
)

NodePanel.propTypes = {
  classes: PropTypes.object.isRequired
}

export default R.compose(
  React.memo,
  withStyles(styles)
)(NodePanel)
