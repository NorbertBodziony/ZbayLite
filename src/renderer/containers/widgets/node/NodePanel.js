import React from 'react'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'

import NodePanelComponent from '../../../components/widgets/node/NodePanel'
import nodeHandlers from '../../../store/handlers/node'
import { useInterval } from '../../hooks'

export const mapDispatchToProps = dispatch => bindActionCreators({
  getStatus: nodeHandlers.epics.getStatus
}, dispatch)

export const NodePanel = ({ className, getStatus }) => {
  useInterval(getStatus, 15000)
  return (
    <NodePanelComponent className={className} />
  )
}

export default connect(null, mapDispatchToProps)(NodePanel)
