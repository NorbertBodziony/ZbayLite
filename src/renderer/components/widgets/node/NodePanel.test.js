import React from 'react'
import { shallow } from 'enzyme'

import { NodePanel } from './NodePanel'
import { mockClasses } from '../../../../shared/testing/mocks'

describe('NodePanel', () => {
  it('renders component', () => {
    const result = shallow(
      <NodePanel classes={mockClasses} />
    )
    expect(result).toMatchSnapshot()
  })
})
