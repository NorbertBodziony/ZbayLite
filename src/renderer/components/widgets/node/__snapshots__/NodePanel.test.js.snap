// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NodePanel renders component 1`] = `
<WithStyles(Grid)
  container={true}
  justify="center"
>
  <WithStyles(Grid)
    item={true}
    xs={true}
  >
    <WithStyles(ExpansionPanel)
      className="expander-mock"
      square={true}
    >
      <WithStyles(ExpansionPanelSummary)
        expandIcon={<pure(ExpandLessIcon) />}
      >
        <Connect(Component) />
      </WithStyles(ExpansionPanelSummary)>
      <WithStyles(ExpansionPanelDetails)
        className="expansionDetails-mock"
      >
        <React.Memo />
      </WithStyles(ExpansionPanelDetails)>
    </WithStyles(ExpansionPanel)>
  </WithStyles(Grid)>
</WithStyles(Grid)>
`;
